pkgs/build-support/binary-cache/make-binary-cache.py

```diff
@@ -10,44 +11,64 @@ def dropPrefix(path, nixPrefix):
     return path[len(nixPrefix + "/") :]
 
 
-def processItem(item, nixPrefix, outDir):
+def processItem(
+    item, nixPrefix, outDir, compression, compressionCommand, compressionExtension
+):
     narInfoHash = dropPrefix(item["path"], nixPrefix).split("-")[0]
 
-    xzFile = outDir / "nar" / f"{narInfoHash}.nar.xz"
-    with open(xzFile, "wb") as f:
+    narFile = outDir / "nar" / f"{narInfoHash}{compressionExtension}"
+    with open(narFile, "wb") as f:
         subprocess.run(
-            f"nix-store --dump {item['path']} | xz -c",
+            f"nix-store --dump {item['path']} {compressionCommand}",
             stdout=f,
             shell=True,
             check=True,
         )
 
     fileHash = (
         subprocess.run(
-            ["nix-hash", "--base32", "--type", "sha256", "--flat", xzFile],
+            ["nix-hash", "--base32", "--type", "sha256", "--flat", narFile],
             capture_output=True,
             check=True,
         )
         .stdout.decode()
         .strip()
     )
-    fileSize = os.path.getsize(xzFile)
+    fileSize = os.path.getsize(narFile)
 
-    finalXzFileName = Path("nar") / f"{fileHash}.nar.xz"
-    os.rename(xzFile, outDir / finalXzFileName)
+    finalNarFileName = Path("nar") / f"{fileHash}{compressionExtension}"
+    os.rename(narFile, outDir / finalNarFileName)
 
     with open(outDir / f"{narInfoHash}.narinfo", "wt") as f:
         f.write(f"StorePath: {item['path']}\n")
-        f.write(f"URL: {finalXzFileName}\n")
-        f.write("Compression: xz\n")
+        f.write(f"URL: {finalNarFileName}\n")
+        f.write(f"Compression: {compression}\n")
         f.write(f"FileHash: sha256:{fileHash}\n")
         f.write(f"FileSize: {fileSize}\n")
         f.write(f"NarHash: {item['narHash']}\n")
         f.write(f"NarSize: {item['narSize']}\n")
-        f.write(f"References: {' '.join(dropPrefix(ref, nixPrefix) for ref in item['references'])}\n")
+        f.write(
+            f"References: {' '.join(dropPrefix(ref, nixPrefix) for ref in item['references'])}\n"
+        )
 
 
 def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--compression", choices=["none", "xz", "zstd"])
+    args = parser.parse_args()
+
+    compressionCommand = {
+        "none": "",
+        "xz": "| xz -c",
+        "zstd": "| zstd",
+    }[args.compression]
+
+    compressionExtension = {
+        "none": "",
+        "xz": ".xz",
```

Comment:
Why were the extension changed from `.nar.xz` to `.xz`? That seems wrong.

Reply:
You're right, that would probably have made more sense. But it seems not to matter since the tests pass.

Reply:
Still makes sense to restore the extension, PR at https://github.com/NixOS/nixpkgs/pull/388173.